src/Spice86.Core/Emulator/LoadableFile/Dos/Exe/ExeLoader.cs

```diff
@@ -53,6 +54,12 @@ public class ExeLoader : DosFileLoader {
             _loggerService.Debug("Exe size: {ExeSize}", exe.Length);
         }
         ExeFile exeFile = new ExeFile(exe);
+        if (!exeFile.Valid) {
+            if (_loggerService.IsEnabled(LogEventLevel.Error)) {
+                _loggerService.Error("Invalid EXE file {file}", file);
+            }
+            throw new UnrecoverableException($"Invalid EXE file {file}");
```

Comment:
Added a check to avoid crashes further down the road, some COM files are distributed as EXE. Maybe we should just try to load any invalid EXE as COM?

Reply:
A sure way to see if it is an EXE is to search for Encoding.ASCII.GetBytes("MZ") at location 0x0.

Reply:
Are you sure about ZM? Currently we check for MZ, and when signature is not that, we fail. I was wondering if we should try to load it as COM in this case instead.

Reply:
ZM is very rare but it's mentionned in some wikis.

Loading it as a COM file if the signature is not present despite the EXE extension is fine.

Reply:
Well I added ZM as a valid signature. Loading as COM would be another PR, concretely apart from a badly named file on my disk I didnt encounter the case with supported games.